inference-engine/src/mkldnn_plugin/utils/bfloat16.hpp

```diff
@@ -139,3 +141,80 @@ class numeric_limits<MKLDNNPlugin::bfloat16_t> {
     static constexpr float_round_style round_style = round_to_nearest;
 };
 } // namespace std
+
+namespace mkldnn {
+namespace impl {
+using namespace mkldnn::impl::utils;
+namespace cpu {
+
+template <cpu_isa_t isa>
+struct bf16_emulation_t {
```

Comment:
The main concern I have about this implementation is the fact that we need explicitly pass auxiliary vector register to the ctor. From usability standpoint this is not very good because you need to care about vector allocation in every particular JIT implementation. 
I would suggest to implement emulator via emitter interface: https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/nodes/common/emitter.h. It encapsulates logic of preserving registers state, so in JIT kernel implementation you don't need to care about this at all. Moreover emitter is a common API we use to jit simple operations in order to enable them for codegen feature.

Reply:
Done